src/cpp/core/include/core/http/TcpIpAsyncClientSsl.hpp

```diff
@@ -238,14 +278,43 @@ class TcpIpAsyncClientSsl
    }
 #endif
 
-private:
-   boost::asio::ssl::context sslContext_;
-   boost::scoped_ptr<boost::asio::ssl::stream<boost::asio::ip::tcp::socket> > ptrSslStream_;
-   std::string address_;
-   std::string port_;
-   bool verify_;
-   std::string certificateAuthority_;
-   boost::posix_time::time_duration connectionTimeout_;
+#ifdef __APPLE__
+   class Keychain : public ICertStore
+   {
+   public:
+      Keychain()
+      {
+         // load all certs from the keychain
+         std::vector<KeychainCertificateData> certs = getKeychainCertificates();
+         for (const auto& cert : certs)
+         {
+            // convert the raw bytes from the keychain into a format that OpenSSL can understand
+            const unsigned char* bytePtr = cert.data.get();
+            X509* x509 = d2i_X509(nullptr, &bytePtr, cert.size);
+            if (x509)
```

Comment:
Does a null result here indicate an error we need to log? Or is this expected in the normal course of operations?

Reply:
Sounds good!